packages/wdio-browserstack-service/src/Percy/PercyBinary.ts

```diff
@@ -0,0 +1,209 @@
+import url from 'node:url'
+import yauzl from 'yauzl'
+import fs from 'node:fs'
+import fsp from 'node:fs/promises'
+import pkg from 'follow-redirects'
+const { https } = pkg
+
+import path from 'node:path'
+import os from 'node:os'
+import { spawn } from 'node:child_process'
+import { PercyLogger } from './PercyLogger.js'
+import type { Options } from '@wdio/types'
+
+class PercyBinary {
+    #hostOS = process.platform
+    #httpPath: any = null
+    #binaryName = 'percy'
+
+    #orderedPaths = [
+        path.join(this.#homedir(), '.browserstack'),
+        process.cwd(),
+        os.tmpdir()
+    ]
+
+    constructor() {
+        const base = 'https://github.com/percy/cli/releases/latest/download'
+        if (this.#hostOS.match(/darwin|mac os/i)) {
+            this.#httpPath = base + '/percy-osx.zip'
+        } else if (this.#hostOS.match(/mswin|msys|mingw|cygwin|bccwin|wince|emc|win32/i)) {
+            this.#httpPath = base + '/percy-win.zip'
+            this.#binaryName = 'percy.exe'
+        } else {
+            this.#httpPath = base + '/percy-linux.zip'
+        }
+    }
+
+    #homedir(): string {
+        if (typeof os.homedir === 'function') {
+            return os.homedir()
+        }
```

Comment:
`os.homedir()` is part of node since v2, so there is no reason to check for it. We can assume it is available.

Reply:
Done